rq/worker.py

```diff
@@ -701,7 +701,18 @@ def fork_work_horse(self, job, queue):
         os.environ['RQ_WORKER_ID'] = self.name
         os.environ['RQ_JOB_ID'] = job.id
         if child_pid == 0:
-            self.main_work_horse(job, queue)
+            try:
```

Comment:
I think we should move the `try/except` block into `main_work_horse()`, it looks much cleaner that way.

Reply:
then I will need to make smth like `main_work_horse_int` to do proper testing and make sure `main_work_horse` always terminates gracefully and not just `perform_job`

like this

```python
    def main_work_horse_int(self, job, queue):
        """This is the entry point of the newly spawned work horse."""
        # After fork()'ing, always assure we are generating random sequences
        # that are different from the worker.
        random.seed()

        self.setup_work_horse_signals()
        self._is_horse = True
        self.log = logger
        self.perform_job(job, queue)

        # os._exit() is the way to exit from childs after a fork(), in
        # contrast to the regular sys.exit()
        os._exit(0)

    def main_work_horse(self, job, queue):
        try:
            self.main_work_horse_int(job, queue)
        except:
            job.ended_at = utcnow()
            exc_info = sys.exc_info()
            exc_string = self._get_safe_exception_string(
                traceback.format_exception(*exc_info)
            )
            self.handle_job_failure(job=job, exc_string=exc_string)
            self.handle_exception(job, *exc_info)
            os._exit(1)
```

is it ok?

Reply:
refactored